spec/parsers/bmp_parser_spec.rb

```diff
@@ -0,0 +1,22 @@
+require 'spec_helper'
+
+describe FormatParser::BMPParser do
+  describe 'is able to parse all the examples of BMP' do
+    Dir.glob(fixtures_dir + '/BMP/*.bmp').each do |bmp_path|
```

Comment:
Almost. Given that we only have 2 specific BMP fixtures, would it maybe make sense to parse them in separate examples and check that the pixel dimensions are correct? We are verifying that they are Integers, which is fine and dandy but they might be `-5` if the parser fails somewhere. We can't verify it in this test since it is automatically created by globbing the fixtures dir.

Reply:
Excellent point. Actually, I found that the height could be a negative number that indicates that the data order is from `ordered from top to bottom` the link is this one http://www.dragonwins.com/domains/getteched/bmp/bmpfileformat.htm#The%20Image%20Header in the section `biHeight `. In the case of a negative number we could actually return the `abs` of it and in the `intrinsics` section that the data is ordered inverse

Reply:
Wow. I only speculated this could happen to prove a point but I could never in my wildest dreams imagine this could **actually** be the case. Yes, I believe it's better to give positive image dimensions to callers regardless of the BMP scan order 😆 amazing gem of a find

Reply:
Actually, the `test2.bmp` file has negative height 😆 so it was serving our case perfectly